blog/2024-03-01-config-entry-reconfigure-step.md

```diff
@@ -0,0 +1,44 @@
+---
+author: G Johansson
+authorURL: https://github.com/gjohansson-ST
+title: "Config Entries can now provide a reconfigure step"
+---
+
+As of Home Assistant Core 2024.4 config entries can now be reconfigured by adding a `reconfigure` step in their config flows.s
+
+This is not to replace optional configuration (`OptionsFlow`) but instead to allow the user to change setup configuration after a config entry has been created.
+
+### Integrations updating existing config entries
+
+Some integrations implements a reconfiguration option by updating an existing config entry if a matching config entry is being created by the user. This is no longer allowed, integrations should be migrated to implement the `reconfigure` step.
+
+### Reconfiguration vs. Reauthentication
+
+The `reconfigure` step does not replace a `reauth` step and they have different purposes.
+
+Reauthentication should be started automatically by the integration in the case a login / token / etc. is invalidated so the user has an option to adjust those settings.
+
+Reconfiguration is started by the user from the config entry options menu and should be implemented to update config entry data which are not optional for the integration to work but is also not used for authentication purposes.
```

Comment:
```suggestion
Reconfiguration is started by the user from the config entry options menu and should be implemented to update config entry data which is not optional for the integration to work but is also not used for authentication purposes.
```

Reply:
Why shouldn't it be authentication data?

Reply:
It could probably be in certain cases so I'll rephrase this.